node_modules/oae-activity/tests/test-email.js

```diff
@@ -135,6 +135,112 @@ describe('Activity Email', function() {
         });
     };
 
+    describe('Templates', function() {
+
+        /**
+         * Test that verifies the state of the activity email footer template based on instance
+         * configuration
+         */
+        it('verify footer template', function(callback) {
+            TestsUtil.generateTestUsers(camAdminRestContext, 3, function(err, users, simong, nico, mrvisser) {
+                assert.ok(!err);
+
+                // Clear emails to start
+                EmailTestsUtil.collectAndFetchAllEmails(function() {
+                    var instanceName = TestsUtil.generateRandomText(1);
+                    var instanceURL = util.format('http://www.instance.oaeproject.org/%s', TestsUtil.generateRandomText(1));
+                    var hostingOrganization = TestsUtil.generateRandomText(1);
+                    var hostingOrganizationURL = util.format('http://www.host.oaeproject.org/%s', TestsUtil.generateRandomText(1));
+
+                    var instanceConfigFields = ['oae-tenants/instance/instanceName', 'oae-tenants/instance/instanceURL'];
+                    var hostingOrganizationConfigFields = ['oae-tenants/instance/hostingOrganization', 'oae-tenants/instance/hostingOrganizationURL'];
+
+                    var updateConfigInstance = {
+                        'oae-tenants/instance/instanceName': instanceName,
+                        'oae-tenants/instance/instanceURL': instanceURL
+                    };
+
+                    var updateConfigHostingOrganization = {
+                        'oae-tenants/instance/hostingOrganization': hostingOrganization,
+                        'oae-tenants/instance/hostingOrganizationURL': hostingOrganizationURL
+                    };
+
+                    ConfigTestUtil.clearConfigAndWait(globalAdminRestContext, null, _.union(instanceConfigFields, hostingOrganizationConfigFields), function(err) {
+                        assert.ok(!err);
+
+                        // Create a link, let Nico manage it
+                        RestAPI.Content.createLink(mrvisser.restContext, 'Google', 'Google', 'public', 'http://www.google.ca', [nico.user.id], [], [], function(err, link) {
+                            assert.ok(!err);
+
+                            // Ensure the email is as expected
+                            EmailTestsUtil.collectAndFetchAllEmails(function(messages) {
+                                assert.notEqual(messages[0].html.indexOf('Apereo <a href=\"http://www.oaeproject.org\" '), -1);
```

Comment:
I'm not sure such a function would be very re-usable outside this test. Can you give an example of what the function signature would be and where it would sit?